orttraining/orttraining/test/training_ops/cuda/gather_elements_grad_test.cc

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "gtest/gtest.h"
+#include "test/providers/provider_test_utils.h"
+
+namespace onnxruntime {
+namespace cuda {
+namespace test {
+
+TEST(GatherElementsGrad, WithoutAxis) {
+  onnxruntime::test::OpTester test("GatherElementsGrad", 1, kMSDomain);
+  std::vector<int64_t> data_shape = {3, 3};
+  test.AddInput<int64_t>("data_shape", {2}, data_shape);
+
+  test.AddInput<int64_t>("indices", {2, 3},
+                         {1, 0, 2,
+                          0, 2, 1});
+
+  // update is dY
+  test.AddInput<float>("update", {2, 3},
+                       {1.0f, 1.1f, 1.2f,
+                        2.0f, 2.1f, 2.2f});
+
+  test.AddOutput<float>("y", {3, 3},
+                        {2.0f, 1.1f, 0.0f,
+                         1.0f, 0.0f, 2.2f,
+                         0.0f, 2.1f, 1.2f});
+  test.Run();
+}
+
+TEST(GatherElementsGrad, WithAxis) {
+  onnxruntime::test::OpTester test("GatherElementsGrad", 1, kMSDomain);
+  test.AddAttribute<int64_t>("axis", 1);
```

Comment:
>axis [](start = 30, length = 4)

another negative axis case ?

Reply:
there is one, GatherElementsGrad, NegativeAxis. Is this what you mean?